src/mono/mono/mini/interp/transform.c

```diff
@@ -1111,6 +1111,20 @@ get_data_item_index (TransformData *td, void *ptr)
 	return index;
 }
 
+static guint16
+get_data_item_index (TransformData *td, void *ptr)
+{
+	guint32 index = get_data_item_wide_index (td, ptr);
+	g_assertf (index < (UINT16_MAX - 1), "Interpreter data item index 0x%x for method '%s' overflows", index, td->method->name);
```

Comment:
I don't understand the reasoning for this comparison. Why not simply `index <= G_MAXUINT16` ? Since provided index must just fit in an guint16 slot.

Reply:
Eh. you're right. For some reason the fact that they're stored as `GUINT_TO_POINTER (index + 1)` in the hash table was throwing me off.  But that doesn't matter since it will be wider there.